src/components/MarkdownPage/MarkdownPage.js

```diff
@@ -43,106 +54,160 @@ const getPageById = (sectionList: Array<Object>, templateFile: ?string) => {
-        title={`${titlePrefix}${titlePostfix}`}
-      />
-      <div css={{flex: '1 0 auto'}}>
-        <Container>
-          <div css={sharedStyles.articleLayout.container}>
-            <Flex type="article" direction="column" grow="1" halign="stretch">
-              <MarkdownHeader title={titlePrefix} />
-
-              {(date || hasAuthors) && (
-                <div css={{marginTop: 15}}>
-                  {date}{' '}
-                  {hasAuthors && (
-                    <span>
-                      by{' '}
-                      {toCommaSeparatedList(authors, author => (
-                        <a
-                          css={sharedStyles.link}
-                          href={author.frontmatter.url}
-                          key={author.frontmatter.name}>
-                          {author.frontmatter.name}
-                        </a>
-                      ))}
-                    </span>
-                  )}
-                </div>
-              )}
+class MarkdownPage extends Component<Props> {
+  ARROW_RIGHT: number = 39;
+  ARROW_LEFT: number = 37;
 
-              <div css={sharedStyles.articleLayout.content}>
-                <div
-                  css={[sharedStyles.markdown]}
-                  dangerouslySetInnerHTML={{__html: markdownRemark.html}}
-                />
+  prevPage: ?Page = getPageById(
+    this.props.sectionList,
+    this.props.markdownRemark.frontmatter.prev,
+  );
+  nextPage: ?Page = getPageById(
+    this.props.sectionList,
+    this.props.markdownRemark.frontmatter.next,
+  );
+
+  navigate = (pageId: string) => {
+    const section = findSectionForPath(
+      this.props.location.pathname,
+      this.props.sectionList,
+    );
+
+    if (section && typeof section.directory === 'string') {
+      navigateTo(slugify(pageId, section.directory));
+    }
+  };
+
+  handleKeyboardNavigation = (event: SyntheticKeyboardEvent<*>) => {
+    switch (event.keyCode) {
+      case this.ARROW_RIGHT:
+        if (this.nextPage) {
+          this.navigate(this.nextPage.id);
+        }
+        break;
+      case this.ARROW_LEFT:
+        if (this.prevPage) {
+          this.navigate(this.prevPage.id);
+        }
+        break;
+    }
+  };
+
+  componentDidMount() {
+    if (this.props.enableKeyboardNavigation) {
+      window.addEventListener('keydown', this.handleKeyboardNavigation);
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.props.enableKeyboardNavigation) {
+      window.removeEventListener('keydown', this.handleKeyboardNavigation);
+    }
+  }
 
-                {markdownRemark.fields.path && (
-                  <div css={{marginTop: 80}}>
-                    <a
-                      css={sharedStyles.articleLayout.editLink}
-                      href={`https://github.com/reactjs/reactjs.org/tree/master/content/${
-                        markdownRemark.fields.path
-                      }`}>
-                      Edit this page
-                    </a>
+  render() {
```

Comment:
No changes in the render method besides the `next` and `prev` variables moved to a class property. The diff is purely cosmetic due to the conversion from a functional component to a class.